tuta-sdk/rust/src/crypto/entity_facade.rs

```diff
@@ -137,27 +127,27 @@ impl EntityFacade {
         }
     }
 
-    fn map_value(&self, entity: ParsedEntity, session_key: &AesKey, key: &String, model_value: &ModelValue) -> Result<(ElementValue, HashMap<String, ElementValue>, HashMap<String, ElementValue>), ApiCallError> {
-        let mut final_ivs: HashMap<String, ElementValue> = Default::default();
+    fn map_value(&self, value: Option<&ElementValue>, session_key: &AesKey, key: &String, model_value: &ModelValue) -> Result<(ElementValue, ElementValue, HashMap<String, ElementValue>), ApiCallError> {
+        let mut final_iv: ElementValue = ElementValue::Bytes(Default::default());
         let mut errors: HashMap<String, ElementValue> = Default::default();
```

Comment:
I think it still stands, I think you could return `Result<(ElementValue, Option<[u8; IV_BYTE_SIZE]>, HashMap<String, ElementValue>), ApiCallError>` and let the caller decide what they do with IV.

Also no need just put empty IV where there is none.

Reply:
I turned final_ivs and errors into optional. I think we still need the errors dict because because during decryption, even with errors we don't crash, we return Ok with the default value and the error. In this case, also, in my opinion error isn't the best name since it's more a warning than an error